src/main/java/frc/team2412/robot/subsystem/IndexSubsystem.java

```diff
@@ -1,11 +1,189 @@
+    // New ColorMatching Instance
+
+    private ColorMatch colorMatcher = new ColorMatch();
+
+    // Define Hardware
+
+    private final MultiplexedColorSensor ingestColorSensor;
+    private final MultiplexedColorSensor feederColorSensor;
+
+    private final WPI_TalonFX ingestMotor;
+    private final WPI_TalonFX feederMotor;
+
+    // States
+
+    private IndexMotorState ingestMotorState;
+    private IndexMotorState feederMotorState;
+
+    private IndexPositionState ingestBallState;
+    private IndexPositionState feederBallState;
+
+    // Constructor
+
+    public IndexSubsystem(WPI_TalonFX firstMotor, WPI_TalonFX secondMotor, MultiplexedColorSensor firstColorSensor,
+            MultiplexedColorSensor secondColorSensor) {
+        this.ingestMotor = firstMotor;
+        this.feederMotor = secondMotor;
+        this.ingestColorSensor = firstColorSensor;
+        this.feederColorSensor = secondColorSensor;
+
+        this.ingestMotor.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
+        this.feederMotor.configSupplyCurrentLimit(MAX_MOTOR_CURRENT);
+
+        colorMatcher.addColorMatch(BLUE_CARGO_COLOR); // might change later to account for opposing team color
+        colorMatcher.addColorMatch(RED_CARGO_COLOR);
+
+        colorMatcher.setConfidenceThreshold(0.9);
+
+        ingestMotorStop();
+        feederMotorStop();
+
+        ingestBallState = HAS_NO_BALL;
+        feederBallState = HAS_NO_BALL;
+
+        setName("IndexSubsystem");
+    }
+
+    // Methods
+
+    /**
+     * Spins first motor inward and updates first motor state
+     */
+    public void ingestMotorIn() {
+        ingestMotor.set(INDEX_IN_SPEED);
+        ingestMotorState = IN;
+    }
+
+    /**
+     * Spins first motor outward and updates first motor state
+     */
+    public void ingestMotorOut() {
+        ingestMotor.set(INDEX_OUT_SPEED);
+        ingestMotorState = OUT;
+    }
+
+    /**
+     * Stops first motor and updates first motor state
+     */
+    public void ingestMotorStop() {
+        if (!ingestSensorHasBallIn()) {
+            ingestMotor.set(0);
+            ingestMotorState = STOPPED;
+        }
+    }
+
+    /**
+     * Spins second motor inward and updates second motor state
+     */
+    public void feederMotorIn() {
+        feederMotor.set(INDEX_IN_SPEED);
+        feederMotorState = IN;
+    }
+
+    /**
+     * Spins second motor outward and updates second motor state
+     */
+    public void feederMotorOut() {
+        feederMotor.set(INDEX_OUT_SPEED);
+        feederMotorState = OUT;
+    }
+
+    /**
+     * Stops second motor and updates second motor state
+     */
+    public void feederMotorStop() {
+        feederMotor.set(0);
+        feederMotorState = STOPPED;
+    }
+
+    /**
+     * Checks if ball is positioned at the first sensor
+     */
+    public boolean ingestSensorHasBallIn() { // also might rename later?
+        return (ingestColorSensor.getProximity() > 1500); // value not 1500, to be determined actual value
```

Comment:
Please add names for the proximity sensor value so it can be changed in one place & update everywhere it's used.